reconcile/test/ocm/aus/test_base.py

```diff
@@ -687,6 +718,269 @@ def test_policy_handler_create_cluster_upgrade(
     )
 
 
+def test_policy_handler_create_cluster_upgrade_with_sts_enabled(
+    cluster_upgrade_policy: ClusterUpgradePolicy,
+    ocm_api: OCMBaseClient,
+    mocker: MockerFixture,
+    secret_reader: SecretReaderBase,
+) -> None:
+    create_upgrade_policy_mock = mocker.patch.object(
+        base, "create_upgrade_policy", autospec=True
+    )
+
+    mock_job_controller = mocker.MagicMock()
+    mocker.patch(
+        "reconcile.aus.base.build_job_controller", return_value=mock_job_controller
+    )
+
+    sts_gate_handler_mock = mocker.patch(
+        "reconcile.aus.version_gates.sts_version_gate_handler.STSGateHandler",
+        autospec=True,
+    )
+    sts_handler_instance = mocker.MagicMock()
+    sts_handler_instance.upgrade_rosa_roles.return_value = True
+    sts_handler_instance.handle.return_value = True
+    sts_handler_instance.gate_applicable_to_cluster.return_value = True
```

Comment:
see if you can use `create_autospec(K8sJobController)` to make it more explicit, same for other `MagicMock` in this file